cvat/apps/engine/views.py

```diff
@@ -2283,6 +2283,15 @@ def perform_update(self, serializer):
                 "Please send a PATCH request with updated parent label data instead.",
                 code=status.HTTP_400_BAD_REQUEST)
 
+        attributes = self.request.data.get('attributes', [])
+        for attribute in attributes:
+            if attribute.get('input_type') == 'checkbox':
+                def_val = attribute.get('default_value')
+                values = attribute.get('values', [])
+                if values and values[0] not in ['true', 'false']:
+                    raise ValidationError('checkbox values must be either true or false')
+                if def_val and def_val not in ['true','false']:
+                    raise ValidationError("default_value for checkbox must be either true or false")
```

Comment:
Hello,

- We also need this check when we create labels, not only when update.
- But code from these views not always executed. For example when we importing a task from a backup, it does not use this code (please, see: `cvat/apps/engine/backup.py:_ImporterBase:_create_labels`). That is why we should move all these Validation checks to AttributeSerializer.
- On the other hand, incorrect values may already be saved in the database. And in this case we need a data migration file to fix the issue. 

The issue is not so trivial as it may look. What do you think?

Reply:
I will proceed with this:
[*] Move the validation logic to 'AttributeSerializer'
[*] Make a data migration to fix existing invalid checkbox values in database.

Thanks for your detailed feedback. If you have furthur suggestions, please let me know.